packages/joyid/src/script-info.ts

```diff
@@ -0,0 +1,269 @@
+import {
+  Cell,
+  QueryOptions,
+  Script,
+  WitnessArgs,
+  blockchain,
+  CellCollector,
+  CellProvider,
+  utils,
+  Address,
+  CellDep,
+  HexString,
+  HashType,
+} from "@ckb-lumos/base";
+import { bytes } from "@ckb-lumos/codec";
+import { parseAddress } from "@ckb-lumos/helpers";
+import { Aggregator, getJoyIDCellDep, getJoyIDLockScript } from "@joyid/ckb";
+import {
+  FromInfo,
+  LockScriptInfo,
+  parseFromInfo,
+} from "@ckb-lumos/common-scripts";
+import { addCellDep } from "@ckb-lumos/common-scripts/lib/helper";
+import { getCotaTypeScript } from "./constants";
+import { Config, getConfig, predefined } from "@ckb-lumos/config-manager";
+
+export class JoyIDCellCollector {
+  readonly fromScript: Script;
+  private readonly cellCollector: CellCollector;
+
+  constructor(
+    private fromInfo: FromInfo,
+    cellProvider: CellProvider,
+    {
+      queryOptions = {},
+      config = getConfig(),
+    }: { queryOptions?: QueryOptions; config?: Config }
+  ) {
+    if (!cellProvider) {
+      throw new Error(
+        `cellProvider is required when collecting JoyID-related cells`
+      );
+    }
+
+    if (typeof fromInfo !== "string") {
+      throw new Error(`Only the address FromInfo is supported`);
+    }
+
+    const { fromScript } = parseFromInfo(fromInfo, { config });
+    this.fromScript = fromScript;
+
+    queryOptions = {
+      ...queryOptions,
+      lock: this.fromScript,
+      type: queryOptions.type || "empty",
+      data: queryOptions.data || "0x",
+    };
+
+    this.cellCollector = cellProvider.collector(queryOptions);
+  }
+
+  async *collect(): AsyncGenerator<Cell> {
+    const joyIdMainnetCodeHash = getJoyIDLockScript(true).codeHash;
+    const joyIdTestnetCodeHash = getJoyIDLockScript(false).codeHash;
+    if (
+      !bytes.equal(this.fromScript.codeHash, joyIdMainnetCodeHash) &&
+      !bytes.equal(this.fromScript.codeHash, joyIdTestnetCodeHash)
+    ) {
+      return;
+    }
+
+    for await (const inputCell of this.cellCollector.collect()) {
+      yield inputCell;
+    }
+  }
+}
+
+type HexStringWithout0x = string;
+
+export type Connection = {
+  pubkey: HexStringWithout0x;
+  // ckb address
+  address: Address;
+  keyType: "main_key" | "sub_key" | string;
+};
+
+export type JoyIDScriptInfoConfig = {
+  aggregator: Pick<Aggregator, "generateSubkeyUnlockSmt">;
+  cellDeps: CellDep[];
+  joyIdLockScriptTemplate: { codeHash: HexString; hashType: HashType };
+  cotaTypeScriptTemplate: { codeHash: HexString; hashType: HashType };
+};
+
+/* c8 ignore next 9*/
+export function getDefaultConfig(isMainnet: boolean): JoyIDScriptInfoConfig {
+  // https://github.com/nervina-labs/cota-sdk-js/blob/f80d04ea532d72cfe7410ea45af6dc583e140edf/README.md?plain=1#L46-L52
+  const aggregatorUrl = isMainnet
+    ? "https://cota.nervina.dev/mainnet-aggregator"
+    : "https://cota.nervina.dev/aggregator";
```

Comment:
What if passing the aggregator endpoint in for extension. When a boolean is in, using the default endpoint. When a string is in, using the custom endpoint.

Reply:
Good point, I'll refactor it

Reply:
I realized that `getDefaultConfig` is a utility function for generating default configs by simply passing network or mainnet param. To customize the config, users can override the config by themself. I've commented it

```ts
const config = {
  ...getDefaultConfig(true),
  aggregator: new Aggregator('/path/to/my/aggregator')
}

createJoyIDScriptInfo(config)
```

Reply:
Got it